### DISCUSSION TITLE
Expectations for Java 20 in 4.27/2023-03

### DISCUSSION THREAD

**Discussion by merks** [eclipse-jdt/eclipse.jdt.core#905] [Q&A] [answered]
I was testing using Java 20 to run the IDE.  The IDE does run and Eclipse does seem to recognize it:

![image](https://user-images.githubusercontent.com/208716/227564282-d100101e-c9c5-43e6-840f-7d39e3d41feb.png)

But it appears JDT cannot use it.

![image](https://user-images.githubusercontent.com/208716/227564958-9aee7750-10ba-4e6b-9095-a271e2e8d329.png)

I guess that's expected, right?

I just hesitate to make JustJ JRE 20 available if none of the released products can actually use it as a JRE...

**Comment by iloveeclipse** [comment, marked as answer]
Please don't do that with 4.27 Eclipse, it requires extra patch installed to recognize & fully support Java 20.
We've merged BETA_JAVA20 branch yesterday to master, so it can be only used for nightly 4.28 builds.

**Comment by merks** [comment]
Thanks for the information.
